Add some examples to the docs (#36)
woocommerce/qit-cli: README.md

### Globally Using `wget`

_(Pro Tip: Opting for the Composer installation method simplifies the process of [updating QIT](https://woocommerce.github.io/qit-documentation/#/cli/getting-started?id=updating-qit) in the future 😉)_
1. Run `wget https://github.com/woocommerce/qit-cli/raw/trunk/qit`
2. Execute `chmod +x qit`
3. Move the file to a directory in your PATH, such as `sudo mv qit /usr/local/bin/qit`
4. Run `qit` to authenticate with your WooCommerce.com Partner Developer account.

## Examples

- `qit run:e2e my-extension` - Runs the WooCommerce Core E2E tests with your extension active.
- `qit run:e2e my-extension --php_version=8.2` - Same as above, but with PHP 8.2.

> - `qit run:e2e my-extension --php_version=8.2 --woocommerce_version=rc --wordpress_version=rc` - Same as above, but with PHP 8.2, and the release candidate versions of WooCommerce and WordPress, if they are available.

Reviewer: Maybe we can also note in here what the default is if an RC isn't available.

Author: Thanks, added

> - `qit run:e2e my-extension --zip` - Runs the tests using your development build zip, that is still unreleased.

Reviewer: Debating on tweaking the wording here a bit, what do you think?

```suggestion
- `qit run:e2e my-extension --zip` - Runs the tests using an unreleased development build zip.
```

Author: Thanks for the suggestion, tweaked

### Globally Using `wget`

_(Pro Tip: Opting for the Composer installation method simplifies the process of [updating QIT](https://woocommerce.github.io/qit-documentation/#/cli/getting-started?id=updating-qit) in the future 😉)_
1. Run `wget https://github.com/woocommerce/qit-cli/raw/trunk/qit`
2. Execute `chmod +x qit`
3. Move the file to a directory in your PATH, such as `sudo mv qit /usr/local/bin/qit`
4. Run `qit` to authenticate with your WooCommerce.com Partner Developer account.

## Examples

- `qit run:e2e my-extension` - Runs the WooCommerce Core E2E tests with your extension active.
- `qit run:e2e my-extension --php_version=8.2` - Same as above, but with PHP 8.2.
- `qit run:e2e my-extension --php_version=8.2 --woocommerce_version=rc --wordpress_version=rc` - Same as above, but with the release candidate versions of WooCommerce and WordPress, or latest stable if none available.
- `qit run:e2e my-extension --zip` - Runs the tests using an unreleased development build zip.
- `qit run:e2e my-extension --additional_wordpress_plugins=gutenberg` - Activates the "Gutenberg" feature plugin in the test environment as well.

> - `qit run:e2e my-extension --optional_features=hpos` - Enables the WooCommerce "High Performance Order Storage" in the test site.

Reviewer: Minor suggestion, but maybe we can link to HPOS here, also added in the word "feature":

```suggestion
- `qit run:e2e my-extension --optional_features=hpos` - Enables the [WooCommerce "High Performance Order Storage"](https://developer.woocommerce.com/roadmap/high-performance-order-storage/) feature in the test site.
```

> You can use these parameter individually or in combination to create different scenarios for your tests. Run `qit run:<test-type> --help` to see all the available options. Different test types will have different options to choose from.

Reviewer: Minor grammar nit:

```suggestion
You can use these parameters individually or in combination to create different scenarios for your tests. Run `qit run:<test-type> --help` to see all the available options. Different test types will have different options to choose from.
```

- [PHPStan Test](https://woocommerce.github.io/qit-documentation/#/test-types/phpstan)
- [API Test](https://woocommerce.github.io/qit-documentation/#/test-types/api)

> - Compatibility Test

Reviewer: Wonder if we should drop this or at least add a section on it to the documentation.

Author: Dropped it

- Integrate QIT as part of your pull requests with our example [GitHub Workflows integration](https://woocommerce.github.io/qit-documentation/#/workflows/getting-started).
- Create and share application passwords with your development team, allowing them to use the QIT CLI without having full access to your WooCommerce.com account.

## Is this tool exclusive to WooCommerce.com Partner Developers?

QIT is an innovative initiative aimed at enhancing the accessibility of automated testing for developers. Our ultimate aspiration is to extend the benefits of this tool to a wider circle of WordPress developers, thereby strengthening the entire ecosystem.

However, as we find ourselves in the preliminary phase of this project, we are currently extending this offering solely to our WooCommerce.com Partner Developers. If you're eager to utilize QIT and want to voice your interest, we welcome you to [submit an issue](https://github.com/woocommerce/qit-cli/issues) in our GitHub repository. We deeply value your feedback as it plays a critical role in shaping the future of our project.

> 

Reviewer: Wording suggestion, I think we can start by selling the tool then move into how we want to further expand it out. So maybe for this section we can say:

```
QIT is an innovative initiative aimed at enhancing the accessibility of automated testing for developers.

In this initial phase, we've started with bringing this to Woo Marketplace Partner Developers. Our ultimate aspiration is to extend the benefits of this tool to a wider circle of WordPress developers, thereby strengthening the entire ecosystem.

If you're eager to utilize QIT and want to voice your interest, we welcome you to [submit an issue](https://github.com/woocommerce/qit-cli/issues) in our GitHub repository. We deeply value your feedback as it plays a critical role in shaping the future of our project.
```

Let me know what you think.

Author: Thanks, changed

- Integrate QIT as part of your pull requests with our example [GitHub Workflows integration](https://woocommerce.github.io/qit-documentation/#/workflows/getting-started).
- Create and share application passwords with your development team, allowing them to use the QIT CLI without having full access to your WooCommerce.com account.

## Is QIT Exclusively Available for WooCommerce.com Partner Developers?

QIT represents a revolutionary step forward in facilitating automated testing for WordPress developers.

In its inaugural phase, we've chosen to primarily target Woo Marketplace Partner Developers. However, our broader vision encompasses extending the availability of this tool to plugins outside of the Marketplace as well, in a commitment to make automated testing accessible to the WordPress ecosystem as a whole.

> Should you be interested in leveraging the potential of QIT and wish to express your support for its broader application, we welcome you to [submit an issue](https://github.com/woocommerce/qit-cli/issues) in our GitHub repository. Your input is invaluable to us, as it greatly contributes to directing the course of the project's future.

Reviewer: Curious on your thoughts for this wording:

```suggestion
If you're interested in leveraging the potential of QIT and want to further support its broader goals, we welcome you to [submit an issue](https://github.com/woocommerce/qit-cli/issues) in our GitHub repository. Your input is invaluable to us, as it greatly contributes to directing the course of the project's future.
```
